src/source/Ice/IceAgent.c

```diff
@@ -1304,10 +1305,24 @@ STATUS incomingDataHandler(UINT64 customData, PSocketConnection pSocketConnectio
         locked = FALSE;
 
         pIceAgent->iceAgentCallbacks.inboundPacketFn(pIceAgent->customData, pBuffer, bufferLen);
-        CHK(FALSE, retStatus);
-    }
+    } else {
+        CHK_STATUS(handleStunPacket(pIceAgent, pBuffer, bufferLen, pSocketConnection, pSrc, pDest,
+                &pStunResponse, &useTurn));
+
+        if (pStunResponse != NULL) {
+            // Release the lock before proceeding
+            MUTEX_UNLOCK(pIceAgent->lock);
+            locked = FALSE;
 
-    CHK_STATUS(handleStunPacket(pIceAgent, pBuffer, bufferLen, pSocketConnection, pSrc, pDest));
+            CHK_STATUS(iceUtilsSendStunPacket(pStunResponse,
+                    (PBYTE) pIceAgent->localPassword,
+                    (UINT32) STRLEN(pIceAgent->localPassword) * SIZEOF(CHAR),
+                    pSrc,
+                    pSocketConnection,
+                    pIceAgent->pTurnConnection,
+                    useTurn));
```

Comment:
dummy question, when do we need to send stun packet without using turn connection?
appears `iceUtilsSendStunPacket` is taking too many parameters.

Reply:
This is a util function that can send stun packets via turn or direct P2P.